### DISCUSSION TITLE
How to enable CUDA acceleration and benchmark  metric of single image pair

### DISCUSSION THREAD

**Discussion by jakubMitura14** [Project-MONAI/MONAI#3510] [Q&A] [answered]
Hello I am comparing Monai metrics as gold standard to some other tools - for this purpose I want to  run the metrics like Dice metric ...  with CUDA acceleration on given images - as an example i took whole body bone segmentation where I compare original to translated image. Metric evaluates but I do not see how to switch on CUDA acceleration for it and synchronize GPU in order to get valid benchmark results.

I had seen this example https://github.com/Project-MONAI/tutorials/blob/master/acceleration/fast_training_tutorial.ipynb where optimizers and transforms are done with CUDA acceleration, but I am not sure what about metrics.

```
import monai
from monai.metrics import Cumulative
dice_metric = DiceMetric(include_background=True, reduction="mean")
dice_metric(y_pred=not_translated[0, :], y=translated[0, :])  # callable to add metric to the buffer
metric = dice_metric.aggregate().item() 
```

**Comment by Nic-Ma** [comment, marked as answer]
Hi @jakubMitura14 ,

Thanks for your interest and experiments here.
Actually, when you compute metrics for the GPU Tensor data(image, label), it executes on GPU.

Thanks.

**Reply by jakubMitura14** [reply]
Thank You do you have an example how to switch normal tensor to gpu tensor?

**Reply by Nic-Ma** [reply]
Hi @jakubMitura14 ,

That's PyTorch feature, just call something like:
```py
a = torch.Tensor([XXX])
a.to(torch.device("cuda"))
```

Thanks.

**Reply by jakubMitura14** [reply]
Thanks it works !
